src/compressor/brotli/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+# brotli
+
+set(brotli_sources 
+  CompressionPluginBrotli.cc
+  BrotliCompressor.cc
+)
+include(ExternalProject)
+ExternalProject_Add(brotli_ext
+  DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/src/
+  GIT_REPOSITORY "https://github.com/google/brotli.git"
+  GIT_TAG "master"
+  SOURCE_DIR ${CMAKE_BINARY_DIR}/src/brotli
+  CONFIGURE_COMMAND ./configure-cmake --disable-debug 
+  INSTALL_COMMAND ""
+  BUILD_COMMAND $(MAKE)
+  BUILD_IN_SOURCE 1
+  INSTALL_COMMAND "")
+
+ExternalProject_Add_Step(brotli_ext forcebuild
+  DEPENDEES configure
+  DEPENDERS build
+  COMMAND "true"
+  ALWAYS 1)
+
+set(bortli_libs brotli::common brotli::dec brotli::enc)
+foreach(lib ${bortli_libs})
+  add_library(${lib} STATIC IMPORTED)
+  add_dependencies(${lib} brotli_ext)
+  STRING(REGEX REPLACE "::" "" static_name ${lib})
+  set_property(TARGET ${lib} PROPERTY IMPORTED_LOCATION "${CMAKE_BINARY_DIR}/src/brotli/lib${static_name}-static.a")
+endforeach()
+
+add_library(ceph_brotli SHARED ${brotli_sources})
+add_dependencies(ceph_brotli ${CMAKE_SOURCE_DIR}/src/ceph_ver.h)
+target_include_directories(ceph_brotli	PRIVATE "${CMAKE_BINARY_DIR}/src/brotli/c/include")
+List(REVERSE bortli_libs)
```

Comment:
why reverse the libs list?

Reply:
Because Depend sequence.

Reply:
nit, if `brotli::dec` and `brotli::enc` depend on `brotli::common`, why not order it the right way at https://github.com/ceph/ceph/pull/19549/files#diff-66716f6999869d6458bd6a9251638b70R25 ?